index.html

```diff
@@ -274,50 +294,67 @@ <h3>Namespace Specific Identifier (NSI)</h3>
-                an analogous keypair on a different curve), and then selecting as the NSI a subset of the public
-                verification key that is at least 128 bits, and that is dependent on <code>keyfmtchar</code>. For
-                <code>keyfmtchar</code> = "1", the NSI = the most significant 16 bytes of the public verification key.</p>
-            <p>In this way, the DID can be known to begin its existence already associated with keys, and the owner
-                of the new DID is guaranteed to be the only entity in the world who possesses the private key.
-                This guarantees the initial integrity of the DID's chain of custody.
-            </p>
+            <p>The unique <em>numeric basis</em> underlying a Peer DID MUST be generated as follows:</p>
+            <ul>
+                <li>Create a <em>genesis</em> version of the DID Doc for the DID. The genesis version must include at
+                least one verification key.</li>
+                <li>Remove the main <code>id</code> field (the one that would normally contain the DID value) from the DID Doc.
+                This creates a variant of the peer DID doc known as the <em>stored variant</em>, as opposed to the
+                <em>resolved variant</em> that would have an actual DID value in the <code>id</code> field.</li>
+                <li>Calculate the SHA256 hash of the bytes of the stored variant of the DID Doc, and select the first 16
+                bytes of the result as the DID's numeric value.</li>
+            </ul>
+            <p>By basing the numeric value of the DID on the genesis version of the DID Doc, the DID can begin its
+                lifecycle with any number of keys and endpoints, and when the doc is signed or auth-encrypted by one of
+                the keys, the recipient can know it has not been modified since creation. The guarantees the initial
+                integrity of the DID's chain of custody.</p>
+            <p>By hashing the stored variant, we avoid the circular problem of
+                including the DID in the data that's being hashed. This means that a peer DID doc must be resolved by
+                converting the stored variant into a resolved variant by inserting the
+                value of the DID being resolved, during resolution.</p>
         </section>
         <section>
             <h3>Examples</h3>
             <p>A convenient regex to match <code>peer</code> DIDs is: <br><br>
-                <code>^did:peer:1:[1-9A-HJ-NP-Za-km-z]{21,22}$</code>.
+                <code>^did:peer:1-F1220[a-fA-F0-9]{8,40}$</code>.
             </p>
-            <p>A valid <code>peer</code> DID might be: <code>did:peer:1:YTEFYzByfU2RwJPyULfLLn</code>.</p>
+            <p>A valid <code>peer</code> DID might be: <code>did:peer:1-F1220479cbc07c3f991725836a3aa2a581ca2029198aa420b9d99bc0e131d9f3e2cbe</code>.
```

Comment:
This DID is a 32 byte hex string but above it is suggested that it be truncated to 16 bytes.

Reply:
Yes, this one is right and the other is a mistake.